src/Controls/src/Build.Tasks/CreateObjectVisitor.cs

```diff
@@ -94,6 +95,39 @@ public void Visit(ElementNode node, INode parentNode)
 				return;
 			}
 
+			// Convert OnIdiomExtension to OnIdiomExtension<T>
+			if (typeref.FullName == "Microsoft.Maui.Controls.Xaml.OnIdiomExtension")
```

Comment:
can't we handle this as a compiled markup extension (the lines just after these ones) ?